arch/risc-v/src/mpfs/mpfs_i2c.c

```diff
@@ -425,6 +425,10 @@ static int mpfs_i2c_init(struct mpfs_i2c_priv_s *priv)
 
       putreg32(MPFS_I2C_CTRL_ENS1_MASK, MPFS_I2C_CTRL);
 
+#ifdef CONFIG_MPFS_COREI2C
```

Comment:
I believe this is correct for hard-ip as well, not just core-i2c

Reply:
And this actually breaks the hard-ip driver, since the semaphore is no longer initialized anywhere. Maybe initialize the semaphore in the original place and reset it here for all i2c implementations?

Reply:
Semaphore init is now back in the original place and semaphore reset is done in mpfs_i2c_init for both type of drivers